src/lib/merkle_list/merkle_list.ml

```diff
@@ -35,6 +39,10 @@ module Make (Input : Inputs_intf) : Make_intf(Input).S = struct
     List.rev (find_path last)
 
   let verify ~init (merkle_list : proof_elem list) underlying_hash =
-    let result_hash = List.fold merkle_list ~init ~f:hash in
-    equal_hash underlying_hash result_hash
+    let hashes =
```

Comment:
Can the `acc` return the list of hashes and the most recent hash. I don't want too many `exn`s appearing in our codebase.

Reply:
Sure.